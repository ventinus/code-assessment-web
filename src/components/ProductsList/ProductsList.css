.products-list {
  margin-top: 20px; }
