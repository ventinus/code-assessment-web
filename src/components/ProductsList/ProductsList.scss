@import '../../scss/variables/mixins.scss';
@import '../../scss/variables/variables.scss';

.products-list {
  margin-top: 20px;
}
