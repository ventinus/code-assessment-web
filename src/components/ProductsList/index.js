import React from 'react'
import PropTypes from 'prop-types'

import './ProductsList.css'

const ProductsList = ({ children }) => (
  <ul className='products-list'>
    {children}
  </ul>
)

ProductsList.propTypes = {
  children: PropTypes.node,
}

export default ProductsList
