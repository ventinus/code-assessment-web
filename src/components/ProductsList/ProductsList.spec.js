import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import ProductsList from '.'

const setup = props => {
  const component = shallow(
    <ProductsList>{props.children}</ProductsList>
  )

  return {
    component: component,
    children: component.children().at(0),
  }
}

describe('ProductsList component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ProductsList>Test Children</ProductsList>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render children', () => {
    const { children } = setup({ children: 'Test Children' })
    expect(children.text()).toMatch(/^Test Children$/)
  })
})
