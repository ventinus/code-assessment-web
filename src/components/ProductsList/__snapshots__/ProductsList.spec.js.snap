// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList component renders correctly 1`] = `
<ul
  className="products-list"
>
  Test Children
</ul>
`;
